### ISSUE TITLE
[a11y]: Treeview expand/collapse control hit area

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#12769]
### Package

@carbon/react

### Browser

Chrome

### Package version

https://carbon-components-react.netlify.app/?path=/story/components-treeview--default

### React version

storybook

### Automated testing tool and ruleset

n/a

### Assistive technology

n/a

### Increase hit area of expand/collapse
The hit area of the expand/collapse control ("twistee") is a small 16x18 pixels area that falls within the larger hit area for selecting the node (which spans the width of the tree).
![image](https://user-images.githubusercontent.com/14298245/205366375-e888c63c-bb4b-4a69-955a-3defa7e2c061.png)
<img width="501" alt="image" src="https://user-images.githubusercontent.com/14298245/205366762-8654bb30-cf78-45bc-b028-be698b3110b0.png">
The most obvious reason to increase the size of the twistee's hit area is to provide sufficient affordance.
However, a second issue is that since the twistees can be indented (in the different levels of the tree) it means that users wishing to expanda multiple sections are forced to navigate not just vertically, but also horizontally in order to hit the differently oriented targets.
If the whole area to the left of the twistee was included in its hit area, then users would be able to click more accurately and consistently.
Besides increasing the horizontal hit area, consideration should also be given to expanding the hit area to include the height of the span. There is no good reason not to also take the vertical space to ensure it is a full 24 pixels high.

### Reproduction/example

https://carbon-components-react.netlify.app/?path=/story/components-treeview--default

### Steps to reproduce

Navigate to https://carbon-components-react.netlify.app/?path=/story/components-treeview--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

```[tasklist]
### Tasks
- [ ] https://github.com/carbon-design-system/carbon/issues/12800
- [ ] https://github.com/carbon-design-system/carbon/issues/13114
```

**Comment by tay1orjones** [comment]
Thanks for opening this! I agree and have added a "Tasks" tasklist section with the actionable steps.

**Comment by laurenmrice** [comment]
Documenting here for further context a slack conversation that was had about hit areas in the Tree view component. 

<img width="1011" alt="Screen Shot 2022-12-09 at 12 15 11 PM" src="https://user-images.githubusercontent.com/43969356/206756819-cedc2cfa-21cc-49a5-9715-45223291e5d9.png">
